main.cpp

```diff
@@ -199,49 +244,69 @@ void *memalign(size_t a, size_t n) {
             spin_unlock(mag->lock);
             return 0;
         }
-    }
+    //}
     mag->offset = i + m;
     spin_unlock(mag->lock);
     p = MAGIC_ADDR + i;
     ((size_t *)p)[-1] = n;
     return p;
 }
 
-void *malloc(size_t n) {
+void *_malloc(size_t n) {
```

Comment:
Ah shucks! MSVC won't let us override its own malloc implementation in the linkage process? That's important to have since C++ STL depends on libc malloc(). Without this, we'd need to explicitly override the STL allocators. I'm going to see if I can work some magic here persuading the compiler to restore this.

Reply:
Yep, that actually was the first thing I was trying to figure out. Haven't checked it out yet, but maybe we could make MSVC not link against binaries defining malloc? Not sure if it's possible though

Reply:
I have a working solution based off this. The trick I used is to override `operator new` and then, to avoid changing `ggml.c` I supply a custom allocated array to its initializer.